pkg/execution/executor/util.go

```diff
@@ -82,3 +82,30 @@ func sortOps(opcodes []*state.GeneratorOpcode) {
 		return opcodes[i].Op < opcodes[j].Op
 	})
 }
+
+// opGroups groups opcodes by their type, ensuring we run `waitForEvent` opcodes
+// first. This is used to ensure that we save wait triggers as soon as possible,
+// as well as capturing expression errors early.
+func opGroups(opcodes []*state.GeneratorOpcode) [][]*state.GeneratorOpcode {
+	var waitForEventGroup []*state.GeneratorOpcode
+	var otherGroup []*state.GeneratorOpcode
```

Comment:
could you make these as a map or a struct?
nested lists are not something to be considered comprehensive.

also this seems like a simple input -> output function, so would be great to have some unit tests on it.

Reply:
Logic here can probably be better encapsulated if a struct is used as well.
https://github.com/inngest/inngest/pull/872/files#diff-ccd0679a9f24c84fef4f5d4d5fe2c32651d23886d85af8448bb968921c7b8a95R962-R980